### PR TITLE
GuiFadeinBitmapCtrl Fix and enhancement

### PR DESCRIPTION
**Pull request by capnlove**
Fixed infinite loop upon fadeout termination.
add console callback when fadeout is done.

### CODE REVIEW

File: engine/source/gui/guiFadeinBitmapCtrl.cc [hunk 1]
```diff
@@ -93,7 +93,12 @@ class GuiFadeinBitmapCtrl : public GuiBitmapCtrl
       {
          // done state
          alpha = fadeoutTime ? 255 : 0;
-         done = true;
+         if (!done) {
```
Comment (dottools) [inline engine/source/gui/guiFadeinBitmapCtrl.cc]: This coding style makes baby Jesus cry.
And inconsistent with the rest of the file. ;)

**Comment by capnlove** [comment]
If you say so!

**Comment by dottools** [comment]
Didn't have to close, easy to fix with another commit which would show up here automatically.

**Comment by capnlove** [comment]
Oh no worries (I ain't butthurt :) )! The way I see it, making this change consistent would require a minor redesign of the control. No time for that atm so I'll just keep it in my fork for now. Maybe resubmit a more thorough revision later on.

**Comment by greenfire27** [comment]
Infinite loops make me cry. If this really does fix such a monster don't wait too long before sharing again :)

**Pull request closed without merging**